infra/DockerRepoManager.py

```diff
@@ -0,0 +1,189 @@
+    self._run_command(remove_command)
+
+    # Remove all previous images
+    rmi_command = ['docker', 'rmi', self.docker_image]
+    self._run_command(rmi_command)
+
+    # Build builder image
+    _build_image(self.project_name)
+    self.checkout_commit(commit)
+    mount_command = [
+        'docker', 'create', '--name', self.TEMP_IMAGE_NAME, self.docker_image
+    ]
+    self._run_command(mount_command)
+
+    # Start the container to be modified
+    start_container = [
+      'docker', 'start', self.TEMP_IMAGE_NAME
+    ]
+    self._run_command(start_container)
+
+    # Remove outdated source repo from container
+    remove_command = [
+        'docker', 'exec', self.TEMP_IMAGE_NAME, 'rm', '-rf', self.src_on_image
+    ]
+    self._run_command(remove_command)
+ 
+    # Copy updated source repo to container
+    copy_command = [
+        'docker', 'cp',
+        os.path.join(self.full_path, '.'),
+        self.TEMP_IMAGE_NAME + ':' + self.src_on_image
+    ]
+    self._run_command(copy_command)
+
+    # Overwrite current image with new container mount
+    commit_command = [
+        'docker', 'commit', self.TEMP_IMAGE_NAME, self.docker_image
+    ]
+    self._run_command(commit_command)
+
+    # Check the command executed correctly
+    image_commit = self.get_image_commit()
+    if not image_commit == commit:
+      raise DockerRepoManagerException(
+          'Docker commit checkout current: %s desired: %s' % (image_commit,
+                                                              commit))
+
+  def get_image_commit(self):
+    """Gets the current commit SHA from the docker image.
+
+    Returns:
+      Commit sha or None on error
+    """
+    get_commit_command = [
+        'docker', 'run', '--rm', '-it', self.docker_image, 'git', '--git-dir',
+        os.path.join(self.src_on_image, '.git'), 'rev-parse', 'HEAD'
+    ]
+
+    out, err = self._run_command(get_commit_command)
+    if err is not None:
+      return None
+    return str(out).rstrip()
+
+  def _get_docker_image_name(self):
+    """Gets the name of the docker build image for specified repo.
+
+    Returns:
+      The name of the docker build image
+    """
+    is_base_image = _is_base_image(self.project_name)
+    if is_base_image:
+      image_project = 'oss-fuzz-base'
+    else:
+      image_project = 'oss-fuzz'
+    return 'gcr.io/%s/%s' % (image_project, self.project_name)
+
+  def _infer_main_repo(self):
+    """ Trys to guess the main repo of the project based on the Dockerfile.
+  
+    Returns:
+      The guessed repo url path
+
+    Raises:
+      NoRepoFoundException: if the repo can't be inferred
+    """
+    if not _check_project_exists(self.project_name):
+      raise NoRepoFoundException(
+          'No project could be found with name %s' % self.project_name)
+    docker_path = _get_dockerfile_path(self.project_name)
+    with open(docker_path, 'r') as fp:
+      for r in fp.readlines():
+        r.lower()
```

Comment:
.lower() does not mutate the string (in fact strings are immutable in python).

you need `r = r.lower()` instead.